hdkeychain/extendedkey_test.go

```diff
@@ -1013,3 +1014,38 @@ func TestZero(t *testing.T) {
 		}
 	}
 }
+
+func TestExtendedKey_MaximumDepth(t *testing.T) {
+
+	extKey, err := hdkeychain.NewMaster([]byte(`abcd1234abcd1234abcd1234abcd1234`), &chaincfg.MainNetParams)
+	if err != nil {
+		t.Error("MaxDepthTest: Failed to produce test fixture key from string")
+		return
+	}
+
+	var i uint8
+	for i = 0; i < math.MaxUint8; i++ {
+		newKey, err := extKey.Child(uint32(1))
+		if err != nil {
+			t.Error("MaxDepthTest: Failed to produce key required for test")
+			return
+		}
+		extKey = newKey
+	}
+
+	noKey, err := extKey.Child(uint32(1))
+	if noKey != nil {
+		t.Error("MaxDepthTest: Deriving 256th key should not succeed")
+		return
+	}
+
+	if _, ok := err.(error); !ok {
```

Comment:
This is not needed.  The check for `err != hdkeychain.ErrDeriveBeyondMaxDepth` below is sufficient.